docs/source/references/internals/graph_breakdown.rst

```diff
@@ -0,0 +1,43 @@
+Pipeline Writing (Breaking Down the Graph)
+------------------------------------------
+
+Overview
+~~~~~~~~
+
+A diagram to show the main abstractions we use to break down a graph as defined when users define a pipelines.
+
+.. image:: ../../_static/images/graph_reader_classes.png
+  :width: 600
+
+The following sections will describe each abstraction in more detail.
+
+Artifact Collection
+~~~~~~~~~~~~~~~~~~~
+The Artifact Collection is the main entrypoint and abstraction for defining the inputs and outputs modules and pipelines.
+    
+The ArtifactCollection can be thought of as a group of artifacts that the user selects and their corresponding graph(s) that will get refactored into reusable components.
+
+The input and output artifacts and variables can come from different user sessions, so the main job of ArtifactCollection is to group the inputs and outputs by session, creating a SessionArtifacts object for each session, and handling inter session dependencies.  
```

Comment:
It is unclear what is meant by "input" and "output".

Reply:
You're right, its not even necessary for this line so I removed the distinction. Its so engrained in my head that I think about "input and output" ....